tests/scenarios/inner_transactions.spec.ts

```diff
@@ -69,11 +69,7 @@ describe('Inner transactions', () => {
           : groupResult.transactions[index].txID(),
       transaction:
         innerTransactionIndex !== undefined
-          ? Transaction.from_obj_for_encoding({
-              ...groupResult.confirmations![index].innerTxns![innerTransactionIndex].txn.txn,
-              gen: groupResult.confirmations![index].txn.txn.gen,
-              gh: groupResult.confirmations![index].txn.txn.gh,
-            })
+          ? groupResult.confirmations![index].innerTxns![innerTransactionIndex].txn.txn
```

Comment:
Do we not need the encoding thing anymore?

Reply:
Yes we don't need anymore:
- In algosdk 2.9, the type that we receive is `EncodedTransaction` => we have to map it to `algosdk.Transaction`
- In algosdk 3, the type is `algosdk.Transaction` and no mapping is needed